### PR TITLE
All mod toggles in separate category

### PR DESCRIPTION
**Pull request by ItsAlxl**
Implements some changes discussed in #14. Creates a "Toggle Mods" category in the mod options that contains the toggles for all toggleable mods. Categories for each individual mod no longer have the toggle in them, and are only created if the mod has additional settings.

Currently, mods that have no settings and are not toggleable do not appear in the mod options at all, but given that users have to specify their modlist separately anyway, that's probably fine. After all, if a mod doesn't have settings and can't be toggled, then there's nothing to configure, and configuring loaded mods is what the mod options menu is for.

ModTogglePreview

In this image, you can see that the mods "Sorted Mission Grid," "Click2Play," and "FullAuto" do not have their own categories because they don't have any options (other than being toggleable). The mouse is hovering over the toggle for "Click2Play," which displays the mod description as its tooltip.

### CODE REVIEW

**Comment by sclu1034** [comment]
> Currently, mods that have no settings and are not toggleable do not appear in the mod options at all, but given that users have to specify their modlist separately anyway, that's probably fine.

I disagree. Since we have practically no non-DMF mods, people are accustomed to the mod options being a complete list of their installed mods.
And when troubleshooting, "Does it appear in the mod options?" is the easiest way to discern between "has loaded but fails elsewhere" vs "failed during initial load".

**Comment by ItsAlxl** [comment]
Pushed a commit to show untoggleable mods in the "Toggle Mods" list. Their toggles are disabled, locked in the "On" position. I'm not sure of a better way that would still list them in load order without messing up the formatting, and this provides a way of communicating that the mod is loaded and can't be turned off. Hovering over a disabled toggle still provides the mod's description as a tooltip, as in the screenshot below.

It also shows mods that are not toggleable and have no options, which DMF currently does not do.

NoConfigMod

File: dmf/localization/dmf.lua [hunk 1]
```diff
@@ -32,13 +32,11 @@ return {
   percent = {
     en = "%%",
   },
-  toggle_mod = {
```
Comment (Aussiemon) [inline dmf/localization/dmf.lua:35]: Revert these localization changes?

Comment (ItsAlxl) [reply dmf/localization/dmf.lua:35]: Well I added `toggle_mods` for the toggle category name and description, which is for toggling more than one mod. Then the old `toggle_mod` wasn't used anywhere anymore, so I removed it. But I'm happy to make changes / revert if that's preferred.

Comment (Aussiemon) [reply dmf/localization/dmf.lua:35]: Ah I see now - missed the s

**Comment by Aussiemon** [comment]
No issues I can find testing it out. Looks good!

**Pull request merged**